script.js

```diff
@@ -0,0 +1,360 @@
+    for (let i = 0; i < processedText.length; i++) {
+        const character = processedText[i];
+        if (/[a-z]/.test(character)) {
+            if (lettersCount[character]) {
+                lettersCount[character]++;
+            } else {
+                lettersCount[character] = 1;
+            }
+        }
+    }
+    return lettersCount;
+}
+
+const lettersObject = countLetters('The quick brown fox jumps over the lazy dog');
+
+console.log(lettersObject);
+
+// 7.Write the getAgeDifference function that finds the difference between the youngest and oldest person.
+// Inside of the getAgeDifference function call the getYoungestPerson  and getOldestPerson functions that you write yourself.
+
+const peopleArray = [
+  {
+    name: 'Adam',
+    age: 20
+  },
+  {
+    name: 'Amanda',
+    age: 5
+  },
+  {
+    name: 'John',
+    age: 75
+  },
+  {
+    name: 'Dave',
+    age: 15
+  }
+]
+
+console.log(getAgeDifference(peopleArray)); // 70
+
+function getYoungestPerson(people) {
+    const userSortedByAge = people.slice().sort(
+        function (firstUser, secondUser) {
+            return firstUser.age - secondUser.age;
+        }
+    )
+    return userSortedByAge[0];
+}
+
+function getOldestPerson(people) {
+    const userSortedByAge = people.slice().sort(
+        function (firstUser, secondUser) {
+            return firstUser.age - secondUser.age;
+        }
+    )
+    return userSortedByAge[userSortedByAge.length - 1];
+}
+
+function getAgeDifference(people) {
+    const youngestPerson = getYoungestPerson(people);
+    const oldestPerson = getOldestPerson(people);
+    return oldestPerson.age - youngestPerson.age;
+}
+
+// 8. You get an array of numbers, return the sum of all of the positives ones.
+// Example [1,-4,7,12] => 1 + 7 + 12 = 20
+// Note: if there is nothing to sum, the sum is default to 0.
+
+function getPositiveSum(arrayOfNumbers) {
+    let sumOfNumbers = 0;
+    for (let i = 0; i < arrayOfNumbers.length; i++) {
+        if (arrayOfNumbers[i] > 0) {
+            sumOfNumbers = sumOfNumbers + arrayOfNumbers[i];
+        }
+    }
+    return sumOfNumbers;
+}
+
+console.log(getPositiveSum([1,-4,7,12]));
+
+// 9. Complete the square sum function so that it squares each number passed into it and then sums the results together.//
+// For example, for [1, 2, 2] it should return 9.
+
+function getSquareSum(arrayOfNumbers) {
+    let squaredSum = 0;
+    for (let i = 0; i < arrayOfNumbers.length; i++) {
+        const squaredNumber = arrayOfNumbers[i] ** 2;
+        squaredSum = squaredSum + squaredNumber;
+    }
+    return squaredSum;
+}
+
+console.log(getSquareSum([1, 2, 2]));
+
+// 10. Write a function to convert a name into initials. This kata strictly takes two words with one space in between them.
+// The output should be two capital letters with a dot separating them.
+// It should look like this:
+// Sam Harris => S.H
+// patrick feeney => P.F
+
+function getInitials(name) {
```

Comment:
You can simplify it quite a bit using template literals
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals